lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -695,6 +695,29 @@ impl SegmentEntry for ProxySegment {
         self.write_segment.get().read().deleted_point_count()
     }
 
+    fn available_vectors_size_in_bytes(&self, vector_name: &str) -> OperationResult<usize> {
+        let wrapped_size = self
+            .wrapped_segment
+            .get()
+            .read()
+            .available_vectors_size_in_bytes(vector_name)?;
+        let wrapped_count = self.wrapped_segment.get().read().available_point_count();
+        let write_size = self
+            .write_segment
+            .get()
+            .read()
+            .available_vectors_size_in_bytes(vector_name)?;
+        let write_count = self.write_segment.get().read().available_point_count();
+        let deleted_points_count = self.deleted_points.read().len();
+        let stored_points = wrapped_count + write_count;
+        let avaliable_points = stored_points.saturating_sub(deleted_points_count);
+        // because we don't know the exact size of deleted vectors, we assume that they are the same avg size as the wrapped ones
+        Ok(
+            ((wrapped_size as u128 + write_size as u128) * avaliable_points as u128
```

Comment:
`u128` is required here because of probability to overflow `usize`. Here we estimate size using proportion. We know `available_vectors_size_in_bytes` from write and wrapped segments and we know externally deleted amount. Calculate as `(wrapper_size + write_size) * (total_count - external_deleted_count) / total_count`

Reply:
How about keeping a u64, and [saturating](https://doc.rust-lang.org/std/primitive.u64.html#method.saturating_mul) up to `u64::MAX`.

Isn't `u64::MAX` is already bigger than the largest amount of data we ever expect?

Reply:
I propose to use here `u128` because here we multiply size and point amount. For instance, we have segment with 100M OpenAI embeddings. It's `1536dim * 4bytes * 100M count = 614GB = 2^39`. Here we may multiply `2^39` by `100M=2^27` and get some about `2^39 * 2^27=2^66` as an intermediate result before deletion